src/authgss_hash.c

```diff
@@ -92,17 +92,13 @@ svc_rpc_gss_cmpf(const struct opr_rbtree_node *lhs,
 	return (1);
 }
 
-void
+static void
 authgss_hash_init()
 {
 	int ix, code = 0;
 
 	mutex_lock(&authgss_hash_st.lock);
 
-	/* once */
-	if (authgss_hash_st.initialized)
-		goto unlock;
-
```

Comment:
Check in the macro is done without any lock! Two threads can make a call to the macro at the same time, find that it is not initialized, and they both call this function. Then they end up initializing hash table one after the other.